src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super();
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden
+      ? this.hidden = false
+      : this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.__proto__ === Herbivore.prototype && !animal.hidden) {
```

Comment:
It's not the best check. Let's use instanceOf instead. In real life, using __proto__ is bad practice